ui/app/src/Utils/Views.elm

```diff
@@ -47,19 +47,43 @@ checkbox name status msg =
         ]
 
 
-validatedFormField : String -> Result ( String, String ) String -> String -> (String -> msg) -> Html msg
-validatedFormField labelText validatedString classes msg =
-    case validatedString of
-        Ok inputValue ->
+validatedField : (List (Attribute msg) -> List (Html msg) -> Html msg) -> String -> String -> (String -> msg) -> msg -> ValidatedField -> Html msg
+validatedField htmlField labelText classes inputMsg blurMsg field =
+    case field.validationState of
```

Comment:
The three different cases in this function differ only by the css classes `has-success`, `has-danger`, or the absence of these two. Would it be possible to just write the HTML once and switch on the css class?